3 - FightClub/Code/FightClub.cs

```diff
@@ -1,23 +1,103 @@
-﻿using ObjectOrientedProblems.Code.Interfaces;
+﻿using ObjectOrientedProblems.Code.Enums;
+using ObjectOrientedProblems.Code.Interfaces;
 using System;
 
 namespace ObjectOrientedProblems.Code
 {
-	public static class FightClubGame
-	{
-		public static IPowerUp PurchasePowerUp()
-		{
-			throw new NotImplementedException();
-		}
+    public static class FightClubGame
+    {
+        public class Powerup : IPowerUp
+        {
+            public int DamageBuff { get; set; } = 1;
+        }
+
+        public class Potion : IPotion
+        {
+            private int _Healing = 2;
+            private PotionType _PotionType = PotionType.Basic;
+
+            public int Healing => _Healing;
+
+            public PotionType Type => _PotionType;
+
+            public void SetType(PotionType potionType) 
+            {
+                if (potionType == PotionType.Greater)
+                {
+                    _Healing = 5;
+                    _PotionType = PotionType.Greater;
+                }
+                else
+                {
+                    _Healing = 2;
+                    _PotionType = PotionType.Basic;
+                }
+            }
+        }
+
+        public class Fighter : IFighter
+        {
+            private int _Health = 10;
+            private int _Damage = 1;
+            private FighterState _State = FighterState.Healthy;
+            public int Health => _Health;
+
+            public int Damage => _Damage;
+
+            public FighterState State => _State;
+
+            public void DrinkPotion(IPotion potion)
+            {
+                if (_Health + potion.Healing >= 10)
+                {
+                    _Health = 10;
+                }
+                else if (_Health > 0)
+                {
+                    _Health += potion.Healing;
+                }
+                Fighterstate();
+            }
+
+            public void PowerUp(IPowerUp powerUp)
+            {
+                _Damage += powerUp.DamageBuff;
+                if (_Damage <= 0)
+                    _Damage = 1;
+            }
+
+            public void TakeDamage(IFighter fighter)
+            {
+                if (_Health > 0)
+                    _Health -= fighter.Damage;
```

Comment:
One of the tests is called `TakeDamage_ShouldNotDropHealthBelowZero`

Your code *can* reduce the health below zero. If health is 1 and .Damage is 5, you will enter a negative state. There should have been some logic here to prevent this from happening.